File: src/main/java/com/example/timetable/util/SpringSecurityUtil.java
package com.example.timetable.util;

import com.example.timetable.config.security.CustomUserDetails;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.security.core.userdetails.UserDetails;

public class SpringSecurityUtil {
    public static Integer getProfileId() {
        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
        CustomUserDetails userDetails = (CustomUserDetails) authentication.getPrincipal();
        return userDetails.getProfileEntity().getId();
    }
    public static String getCurrentUserId() {
        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
        String username = authentication.getName();
        return username;
    }

    public static UserDetails getUserDetails() {
        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
        return userDetails;
    }
}

File: src/main/kotlin/com/example/w3schooluz/util/SpringSecurityUtil.kt
package com.example.w3schooluz.util


import com.example.w3schooluz.config.CustomUserDetails
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.security.core.userdetails.UserDetails


object SpringSecurityUtil {
    val profileId: Int
        get() {
            val authentication = SecurityContextHolder.getContext().authentication
            val userDetails = authentication.principal as CustomUserDetails
            return userDetails.getProfileEntity()?.id!!
        }
    val currentUserId: String
        get() {
            val authentication =
                SecurityContextHolder.getContext().authentication
            return authentication.name
        }
    val userDetails: UserDetails
        get() {
            val authentication =
                SecurityContextHolder.getContext().authentication
            return authentication.principal as UserDetails
        }
}
